### PR TITLE
[Cortx-dev] Eos 14633 SSL certificate alert displays incorrect time zone

### PR DESCRIPTION
**Pull request by navalkp**
Signed-off-by: Naval Patel <naval.patel@seagate.com>

# Backend

## Problem Statement
<pre>
Story Ref (if any):https://jts.seagate.com/browse/EOS-14633
local timezone not printed in date
</pre>
## Unit testing on RPM done
<pre>
No
</pre>
## Problem Description
<pre>
Local timezone of server is not in date  
</pre>
## Solution
<pre>
Added local timezone in date format
</pre>
## Unit Test Cases
<pre>
Tested code snippet independently 

Output date format is 
2020-10-27 04:14:48 UTC
2020-10-27 09:44:48 IST
</pre>
Signed-off-by:

### CODE REVIEW

File: cicd/pyinstaller/requirment.txt [hunk 1]
```diff
@@ -26,3 +26,4 @@ cryptography==2.8
 python-crontab==2.5.1
 confluent-kafka==1.5.0
 netifaces==0.10.9
+tzlocal==2.1
```
Comment (ujjwalpl) [inline cicd/pyinstaller/requirment.txt:29]: Why cant we use datetime.timezone ? 
https://docs.python.org/3/library/datetime.html

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: cicd/pyinstaller/requirment.txt (hunk 1 above)
Comment (navalkp) [reply cicd/pyinstaller/requirment.txt:29]: I need local timezone info. Datetime does not give me that info.

Comment (navalkp) [reply cicd/pyinstaller/requirment.txt]: removed tzlocal, don't need local timezone , astimezone() with parameter defaults to localtimezone.

File: csm/core/services/security.py [hunk 2]
```diff
@@ -311,16 +311,24 @@ async def _timer_task(self, handler, start: datetime, interval: timedelta):
             current = datetime.now(timezone.utc)
             start = start + interval
 
+    async def _local_timezone(self, from_time):
+        """
+        Convert utc aware time to local timezone and print in specific format
+        """
+        local_time = from_time.astimezone()
```
Comment (kupranay) [inline csm/core/services/security.py:318]: this is not the local time.. this is converted time with  timezone. Please rename variable accordingly

File: csm/core/blogic/const.py [hunk 3]
```diff
@@ -90,6 +90,9 @@
 # File Collector
 BUNDLE_FILE = 'files.tgz'
 
+# Security
+CERT_TIME_FORMAT = "%Y-%m-%d %H:%M:%S %Z%z"
```
Comment (kupranay) [inline csm/core/blogic/const.py:94]: i think we dont need "%z" 
output as 
>>> local_time.strftime("%Y-%m-%d %H:%M:%S %Z%z")
'2020-10-27 22:02:12 IST+0530'
>>> local_time.strftime("%Y-%m-%d %H:%M:%S %Z")
**'2020-10-27 22:02:12 IST'**

only IST should be enough

File: csm/core/services/security.py [hunk 4]
```diff
@@ -311,16 +311,24 @@ async def _timer_task(self, handler, start: datetime, interval: timedelta):
...
+        return local_time.strftime(const.CERT_TIME_FORMAT)
+
     async def _check_certificate_expiry_time(self, current_time):
         warning_days = Conf.get(const.CSM_GLOBAL_INDEX, "SECURITY.ssl_cert_expiry_warning_days")
         try:
             expiry_time = await self.get_certificate_expiry_time()
             expiry_time = expiry_time.replace(tzinfo=timezone.utc)
+            local_time = self._local_timezone(expiry_time)
```
Comment (kupranay) [inline csm/core/services/security.py:326]: Thisi s not local time. Please rename variable accordingly

**Review by kupranay — CHANGES_REQUESTED** [review]

File: csm/core/services/security.py (hunk 2 above)
Comment (navalkp) [reply csm/core/services/security.py]: renamed variable

File: csm/core/blogic/const.py (hunk 3 above)
Comment (navalkp) [reply csm/core/blogic/const.py]: Yes, removed %z

File: csm/core/services/security.py (hunk 4 above)
Comment (navalkp) [reply csm/core/services/security.py]: renamed variable

**Review by kupranay — APPROVED** [review]

**Review by ujjwalpl — APPROVED** [review]

**Pull request merged**